src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesLambdas.java

```diff
@@ -0,0 +1,19 @@
+package com.puppycrawl.tools.checkstyle.checks.coding.unnecessaryparentheses;
+
+public class InputUnnecessaryParenthesesLambdas {
+    int foo(int y) {
+        MathOperation addition = (x) -> x + x;
+        MathOperation addition2 = (x) -> { return x + x; };
+        MathOperation addition3 = x -> x + x;
+        MathOperation2 addition4 = (a, b) -> a + b;
```

Comment:
please make a case like `() -> ....` and `(A a, B b) -> .....`
there are even more interesting cases like `t1 -> t2 -> apply(t1, t2);` (see it [there](http://rveach.no-ip.org/checkstyle/regression/reports/99/Vavr/xref/Vavr/vavr/src-gen/main/java/io/vavr/CheckedFunction2.java.html#L160))
We need more cases of singleline-nested lambdas, please review vavr project, they love lambdas.